### PR TITLE
jwt login function added

### PR DESCRIPTION
**Pull request by shubhwale**

### CODE REVIEW

File: src/app/auth/auth.interceptor.ts [hunk 1]
```diff
@@ -12,14 +12,15 @@ export class AuthInterceptor implements HttpInterceptor {
...
-                headers : req.headers.set('Authorization','Bearer '+localStorage.getItem('token'))
+                headers : req.headers.set('Authorization','Bearer '+localStorage.getItem('accessToken'))
             });
             return next.handle(clonedReq).pipe(
                 tap(
                     succ => {},
                     err => {
                         if(err.status == 401) {
-                            localStorage.removeItem('token');
+                            localStorage.removeItem('AccessToken');
```
Comment (ketannaik) [inline src/app/auth/auth.interceptor.ts:22]: Keep name consistent. AccessToken or accessToken either of them.

File: src/app/shared/user.service.ts [hunk 2]
```diff
@@ -7,15 +7,19 @@ import { HttpClient, HttpHeaders } from "@angular/common/http";
 export class UserService {
```
Comment (ketannaik) [inline src/app/shared/user.service.ts:7]: Why the form builder and form model are defined inside the Service. Should it be in the User Component?

File: src/app/auth/auth.interceptor.ts (hunk 1 above)
Comment (shubhwale) [reply src/app/auth/auth.interceptor.ts]: changes done

File: src/app/shared/user.service.ts (hunk 2 above)
Comment (shubhwale) [reply src/app/shared/user.service.ts:7]: changes done

File: src/app/shared/user.service.ts [hunk 3]
```diff
@@ -1,55 +1,36 @@
...
+  constructor(private http: HttpClient, private user : UserComponent) { }
+  private readonly BaseUri = 'https://localhost:44335/api';
 
   register() {
     var body = {
-      UserName : this.formModel.value.UserName,
-      Email : this.formModel.value.Email,
-      FullName : this.formModel.value.FullName,
-      Password : this.formModel.value.Passwords.Password,
+      Name : this.user.formModel.value.Name,
```
Comment (ketannaik) [inline src/app/shared/user.service.ts:14]: Again, why this user component is defined in the User Service?
Please declare all these objects in component and send that object to the service. 

Service does not have to object creation logic

File: src/app/user/registration/registration.component.ts [hunk 4]
```diff
@@ -9,31 +10,20 @@ import { ToastrComponentlessModule, ToastrService } from 'ngx-toastr';
...
 export class RegistrationComponent implements OnInit {
 
-  constructor(public service : UserService,private toastr : ToastrService) { }
+  constructor(public service : UserService,private toastr : ToastrService, public user : UserComponent) { }
 
   ngOnInit(): void {
   }
 
   onSubmit() {
     this.service.register().subscribe(
```
Comment (ketannaik) [inline src/app/user/registration/registration.component.ts:19]: Technically, your registration object should be created in this file. Once it is created, send it to the service function as a parameter.

File: src/app/shared/user.service.ts (hunk 3 above)
Comment (shubhwale) [reply src/app/shared/user.service.ts:14]: done

File: src/app/user/registration/registration.component.ts (hunk 4 above)
Comment (shubhwale) [reply src/app/user/registration/registration.component.ts:19]: done

**Pull request merged**